partner_relations/model/res_partner.py

```diff
@@ -0,0 +1,316 @@
+                        not in value[2]:
+                    relation_data = relation_obj.read(
+                        cr, uid, [value[1]], ['type_selection_id'],
+                        context=context)[0]
+                    value[2]['type_selection_id'] =\
+                        relation_data['type_selection_id']
+                relation_obj.write(
+                    cr, uid, value[1], value[2], context=context2)
+            if value[0] == 2:
+                relation_obj.unlink(cr, uid, value[1], context=context2)
+
+    def _search_relation_id(
+            self, cr, uid, dummy_obj, name, args, context=None):
+        result = []
+        for arg in args:
+            if isinstance(arg, tuple) and arg[0] == name:
+                if arg[1] not in ['=', '!=', 'like', 'not like', 'ilike',
+                                  'not ilike', 'in', 'not in']:
+                    raise exceptions.ValidationError(
+                        _('Unsupported search operand "%s"') % arg[1])
+
+                relation_type_selection_ids = []
+                relation_type_selection = self\
+                    .pool['res.partner.relation.type.selection']
+
+                if arg[1] == '=' and isinstance(arg[2], (long, int)):
+                    relation_type_selection_ids.append(arg[2])
+                elif arg[1] == '!=' and isinstance(arg[2], (long, int)):
+                    type_id, is_inverse = (
+                        relation_type_selection.browse(arg[2])
+                        .get_type_from_selection_id()
+                    )
+                    result = OR([
+                        result,
+                        [
+                            ('relation_all_ids.type_id', '!=', type_id),
+                        ]
+                    ])
+                    continue
+                else:
+                    relation_type_selection_ids = relation_type_selection\
+                        .search(
+                            cr, uid,
+                            [
+                                ('type_id.name', arg[1], arg[2]),
+                                ('record_type', '=', 'a'),
+                            ],
+                            context=context)
+                    relation_type_selection_ids.extend(
+                        relation_type_selection.search(
+                            cr, uid,
+                            [
+                                ('type_id.name_inverse', arg[1], arg[2]),
+                                ('record_type', '=', 'b'),
+                            ],
+                            context=context))
+
+                if not relation_type_selection_ids:
+                    result = AND([result, [FALSE_LEAF]])
+
+                for relation_type_selection_id in relation_type_selection_ids:
+                    type_id, is_inverse = (
+                        relation_type_selection_id.get_type_from_selection_id()
```

Comment:
AttributeError: 'int' object has no attribute 'get_type_from_selection_id'
relation_type_selection_ids is a list of integers (old API search method returns a list of integers), so you new to use browse, isn't it?